src/OpenApi/ClientGenerator.php

```diff
@@ -388,6 +397,43 @@ protected function addBodyParam(array $arguments, array &$code): void
         }
     }
 
+    protected function addFormData(array $arguments, array &$code): void
+    {
+        $formData = [];
+        $multipart = [];
+
+        foreach ($arguments as $argument) {
+            if ($argument['in'] != 'formData') {
+                continue;
+            }
+
+            if ($argument['is_file']) {
+                $multipart[] = '                [';
+                $multipart[] = sprintf('                    \'name\' => \'%s\',', $argument['name']);
+                $multipart[] = sprintf('                    \'contents\' => fopen($%s, \'r\'),', $argument['name']);
+                $multipart[] = '                ],';
+            } else {
+                $formData[] = sprintf('                \'%s\' => $%s,', $argument['name'], $argument['name']);
+            }
+        }
+
+        if ($formData) {
+            $code[] = '            \'form_params\' => [';
+            foreach ($formData as $argument) {
+                $code[] = $argument;
+            }
+            $code[] = '            ],';
+        }
+
+        if ($multipart) {
+            $code[] = '            \'multipart\' => [';
```

Comment:
In my test the file upload did not work. This is because the `$options` constructed in a `Client` method is not just passed to Guzzle. `BaseClient::prepareAndExecuteRequest()` needs to be modified for this.

Reply:
possible. only just writed the code, havent had the opertunity to test it.